examples/tonic/src/server.rs

```diff
@@ -0,0 +1,96 @@
+use tonic::{transport::Server, Request, Response, Status};
+
+use hello_world::greeter_server::{Greeter, GreeterServer};
+use hello_world::{HelloReply, HelloRequest};
+use opentelemetry::api::{self, HttpTextFormat, KeyValue, Provider};
+use opentelemetry::sdk::Sampler;
+use opentelemetry::{global, sdk};
+use tracing_opentelemetry::{OpenTelemetryLayer, OpenTelemetrySpanExt};
+use tracing_subscriber::{Layer, Registry};
+
+pub mod hello_world {
+    tonic::include_proto!("helloworld"); // The string specified here must match the proto package name
+}
+
+#[derive(Debug, Default)]
+pub struct MyGreeter {}
+
+#[tonic::async_trait]
+impl Greeter for MyGreeter {
+    async fn say_hello(
+        &self,
+        request: Request<HelloRequest>, // Accept request of type HelloRequest
+    ) -> Result<Response<HelloReply>, Status> {
+        // Return an instance of type HelloReply
+        println!("Got a request: {:?}", request);
+
+        let reply = hello_world::HelloReply {
+            message: format!("Hello {}!", request.into_inner().name), // We must use .into_inner() as the fields of gRPC requests and responses are private
+        };
+
+        Ok(Response::new(reply)) // Send back our formatted greeting
+    }
+}
+
+fn tracing_init() -> Result<(), Box<dyn std::error::Error>> {
+    let builder = opentelemetry_jaeger::Exporter::builder()
+        .with_agent_endpoint("127.0.0.1:6831".parse().unwrap());
+
+    let exporter = builder
+        .with_process(opentelemetry_jaeger::Process {
+            service_name: "server".to_string(),
+            tags: vec![KeyValue::new("version", "0.1.0")],
+        })
+        .init()?;
+
+    let provider = sdk::Provider::builder()
+        .with_simple_exporter(exporter)
+        .with_config(sdk::Config {
+            default_sampler: Box::new(Sampler::Always),
+            ..Default::default()
+        })
+        .build();
+
+    let tracer = provider.get_tracer("my-tracer");
+    let opentelemetry = OpenTelemetryLayer::with_tracer(tracer);
+
+    let subscriber = opentelemetry.with_subscriber(Registry::default());
```

Comment:
nit: generally, i find `SubscriberExt::with` a little more idiomatic than `Layer::with_subscriber`, and tend to prefer it in examples where possible:
```suggestion
    let subscriber = Registry::default()
        .with(opentelemetry);
```

It _may_ also be worth showing how we can use the `OpenTelemetry layer in conjunction with other layers, if this is intended as a full-featured example app?

```suggestion
    let subscriber = Registry::default()
         // add the OpenTelemetry subscriber layer
        .with(opentelemetry)
         // add a logging layer
        .with(tracing_subscriber::fmt::Layer::default())
         // add RUST_LOG-based filtering
        .with(tracing_subscriber::EnvFilter::from_default_env());
```
or something? But, feel free to ignore that if you'd prefer to focus on OpenTelemetry.